core/arch/arm/pta/system.c

```diff
@@ -620,13 +620,13 @@ static TEE_Result call_ldelf_dlopen(struct user_ta_ctx *utc, TEE_UUID *uuid,
 	usr_stack -= ROUNDUP(sizeof(*arg), STACK_ALIGNMENT);
 	arg = (struct dl_entry_arg *)usr_stack;
 
-        res = tee_mmu_check_access_rights(utc, TEE_MEMORY_ACCESS_READ |
-                                          TEE_MEMORY_ACCESS_ANY_OWNER,
-                                          (uaddr_t)arg, sizeof(*arg));
-        if (res) {
-                EMSG("ldelf stack is inaccessible!");
-                return res;
-        }
+	res = tee_mmu_check_access_rights(utc, TEE_MEMORY_ACCESS_READ |
+					  TEE_MEMORY_ACCESS_ANY_OWNER,
```

Comment:
I think you can (and should) drop `TEE_MEMORY_ACCESS_ANY_OWNER` here. (would apply also to `call_ldelf_dlsym()`)